pkg/scannerv4/client/client.go

```diff
@@ -33,12 +33,12 @@ type Scanner interface {
 	// GetOrCreateImageIndex first attempts to get an existing index report for the
 	// image reference, and if not found or invalid, it then attempts to index the
 	// image and return the generated index report if successful, or error.
-	GetOrCreateImageIndex(ctx context.Context, ref name.Digest, auth authn.Authenticator) (*v4.IndexReport, error)
+	GetOrCreateImageIndex(ctx context.Context, ref name.Digest, opt ImageRegistryOpt) (*v4.IndexReport, error)
 
 	// IndexAndScanImage scans an image for vulnerabilities. If the index report
 	// for that image does not exist, it is created. It returns the vulnerability
 	// report.
-	IndexAndScanImage(context.Context, name.Digest, authn.Authenticator) (*v4.VulnerabilityReport, error)
+	IndexAndScanImage(context.Context, name.Digest, authn.Authenticator, bool) (*v4.VulnerabilityReport, error)
```

Comment:
Wouldn't it also make more sense to use the same options pattern from `GetOrCreateImageIndex()` here?